### PR TITLE
Get frame delay before advancing to show the current frame for the correct amount of time

### PR DESCRIPTION
**Pull request by TWiStErRob**
See https://github.com/bumptech/glide/issues/1060#issuecomment-201023023
Confirmed working by manually testing with the image I created from the original.

### CODE REVIEW

File: third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/GifDecoder.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public void advance() {
      */
     public int getDelay(int n) {
         int delay = -1;
-        if ((n >= 0) && (n < header.frameCount)) {
+        if (0 <= n && n < header.frameCount) {
```
Comment (sjudd) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/GifDecoder.java]: nit: I prefer the other ordering for <= or ==, it's easier for me to read "n is greater than or equal to 0" than it is "0 is less than or equal to n".

Comment (TWiStErRob) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/GifDecoder.java]: How about "`n` is *between* 0 and `frameCount`"?
It's literally between the two bounds in the code as well.

Comment (sjudd) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/GifDecoder.java]: Yeah I'd prefer 0 <= n < frameCount if the language supported it, but as is, I'm used to reading variable <=> number.

**Comment by TWiStErRob** [comment]
nit fixed

**Comment by sjudd** [comment]
Thanks!

**Pull request merged**